src/GraphQL.Tests/Utilities/SchemaVisitorTests.cs

```diff
@@ -91,5 +94,234 @@ public override void VisitField(FieldType field)
+            }
+
+            public override void VisitObjectGraphType(IObjectGraphType type)
+            {
+                type.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitObjectGraphType(ObjectGraphType type)
+            {
+                type.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitEnum(EnumerationGraphType type)
+            {
+                type.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitEnumValue(EnumValueDefinition value)
+            {
+                value.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitScalar(ScalarGraphType scalar)
+            {
+                scalar.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitField(FieldType field)
+            {
+                field.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitInterface(InterfaceGraphType interfaceDefinition)
+            {
+                interfaceDefinition.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitUnion(UnionGraphType union)
+            {
+                union.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitArgumentDefinition(QueryArgument argument)
+            {
+                argument.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitInputObject(InputObjectGraphType type)
+            {
+                type.Description = GetArgument("description", string.Empty);
+            }
+
+            public override void VisitInputFieldDefinition(FieldType value)
+            {
+                value.Description = GetArgument("description", string.Empty);
+            }
+        }
+
+        public class TestType
+        {
+            public int Id { get; set; }
+
+            public string Name { get; set; }
+        }
+
+        public class TestTypeForUnion
+        {
+            public int Field { get; set; }
+        }
+
+        [Fact]
+        public void can_apply_custom_directive_to_schema()
+        {
+            var schema = new Schema();
+            schema.AddDirective("test", new RegisterTypeDirectiveVisitor());
+
+            var directivesMetadata = schema.GetDirectives();
+            directivesMetadata.ShouldNotBeNull();
+            directivesMetadata.Count.ShouldBe(1, "Only 1 directive should be added");
+            directivesMetadata.ContainsKey("test").ShouldBeTrue();
+
+            schema.FindType("TestType").ShouldNotBeNull();
+        }
+
+        [Fact]
+        public void can_create_custom_directive_for_all_locations()
```

Comment:
Please add directives and tests for those locations too:

SCHEMA
SCALAR
INPUT_OBJECT
INPUT_FIELD_DEFINITION

Reply:
Add test for this locations.
Note that currently there is no support of `ScalarTypeDefinition` in schema first approach. So test only for graph-type first.